cmd/user/internal/domain/user/commands.go

```diff
@@ -34,6 +37,24 @@ func NewCommandFromPayload(contract string, payload []byte) (domain.Command, err
 	case RegisterUserWithEmail:
 		registerWithEmail := RegisterWithEmail{}
 		err := unmarshalPayload(payload, &registerWithEmail)
+		// validation rules
```

Comment:
I dont like it here, the way i would do it is as I said earlier create value objects

```go
type RegisterWithEmail struct {
	Email E EmailAddress ail `json:"email"`
}

type EmailAddress string

func (e EmailAddress) UnmarshalJSON(b []byte) error {
	err := json.Unmarshal(b, &nb.Bool)
	if err != nil {
		return errors.Wrap(err, errors.INTERNAL, Unmarshal error")
	}

	if !govalidator.MapData.IsEmail {
		return errors.New(errors.INTERNAL, "Invalid email address")
	}

	return nil
}
```

This way method will be called when doing `json.Unmarshal`

Reply:
No worries, not sure I fully understand it yet but we can work it out later.